### DISCUSSION TITLE
Syntax Highlighting Issue (SourceRequest)

### DISCUSSION THREAD

**Discussion by krmbzds** [mfussenegger/nvim-dap#495] [General]
Hello,

I use [nvim-dap](https://github.com/mfussenegger/nvim-dap) along with [nvim-dap-ruby](https://github.com/suketa/nvim-dap-ruby) to debug Ruby code. With release `1.5.0`  of [debug.rb](https://github.com/ruby/debug), syntax highlighting stopped working for me. I reported this in [release issues](https://github.com/ruby/debug/issues/598#issuecomment-1083703639) of debug.rb and got some feedback regarding nvim-dap.

Quoting from **https://github.com/ruby/debug/issues/598#issuecomment-1083982077**:
> * nvim dap support seems no coloring on src acquired by [Source request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Source) (VSCode does). I think it should be fixed on nvim dap support (*1)
> * To cancel `Source` request, `localfs` configuration should be added in launch.json on VSCode. This configuration is passed to DAP (debug.gem) by `launch` or `attach` request arguments (VSCode pass the configuration in arguments). But it is implementation defined way, and not sure nvim DAP support can pass these parameters.
>
> * We can make further heuristics, for example assume `localfs = true` if host is localhost. But I'm not sure we can cover all of them (for example, 127.0.0.1 is not only local node IP address).
> 
> (*1) For example, `eval`'ed code should be acquired by `Source` request, so fixing it will help.

I started this discussion to pass-on the suggestions made by @ko1 as I do not have enough knowledge on either `nvim-dap` or `debug.rb` to open a proper issue for this. If anyone's generous enough to explain what's going on or suggest a workaround for this I'm all ears.

**Comment by mfussenegger** [comment]
This is probably related to https://github.com/mfussenegger/nvim-dap/issues/334
